text/0965-reactivity.md

```diff
@@ -0,0 +1,419 @@
+---
+stage: accepted
+start-date: 2023-09-20T00:00:00.000Z # In format YYYY-MM-DDT00:00:00.000Z
+release-date: # In format YYYY-MM-DDT00:00:00.000Z
+release-versions:
+teams: # delete teams that aren't relevant
+  - data
+  - framework
+  - learning
+  - steering
+  - typescript
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/965 
+project-link:
+suite: 
+---
+
+<!--- 
+Directions for above: 
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+suite: Leave as is
+-->
+
+# Rethinking Reactivity
+
+## Summary
+
+Leading up to the Octane Edition, we laid out a plan to get rid of `<object>.get` and `<object>.set` in favor of `@tracked` class properties, which paved the way for writing reactive code in a way that is _the most_ "just javascript" out of our framework peers at the time,
+
+Additionally, this was taken as an opportunity to move from push-based reactivity to pull-based reactivity, and lead to a long exploration of patterns, trial-and-error, and developing new patterns to achieve easier testing, debugging and maintenance.  But most of this happened long after the official documentation for Octane had landed -- and even at the time of writing this RFC, the most important concepts about reactivity would fall well under the ["In-depth topics"](https://guides.emberjs.com/release/in-depth-topics/autotracking-in-depth/) section of the documentation. 
+
+This RFC aims for two things:
+- focus on teaching reactivity concepts, so that folks can be better equipped for what would be "advanced concepts" today.
+- defining and bringing cohesion to our reactive primitives so that the above is much easier to achieve. 
+  - This may necessitate additional sub-RFCs. 
+
+## Motivation
+
+The ember ecosystem has a number of "third-party" abstractions for some core reactive primitives, that could benefit (from teaching, performance, and cohesion perspectives) from being pulled in to a cohesive set of utilities from one source. This will help with longer-term goals of [simplifying imports](https://github.com/emberjs/rfcs/pull/946) -- today we have a _pit of exhaustion_ when it comes to how many imports are needed to make components.
```

Comment:
```suggestion
The ember ecosystem has a number of "third-party" abstractions for some core reactive primitives, that could benefit (from teaching, performance, and cohesion perspectives) from being pulled into a cohesive set of utilities from one source. This will help with longer-term goals of [simplifying imports](https://github.com/emberjs/rfcs/pull/946) -- today we have a _pit of exhaustion_ when it comes to how many imports are needed to make components.
```